From: Matthieu Gallien <matthieu_gallien@yahoo.fr>
Date: Sun, 22 Sep 2019 18:47:59 +0200
Subject: fix compatibility with Qt5.11

Summary: Kirigami2 requires Qt 5.11 but D22974 broke that promises

Test Plan: Elisa runs fine against Qt5.11

Reviewers: #kirigami, mart

Reviewed By: #kirigami, mart

Subscribers: broulik, plasma-devel

Tags: #kirigami

Differential Revision: https://phabricator.kde.org/D24147
---
 src/controls/Action.qml            | 2 +-
 src/controls/ActionToolBar.qml     | 4 ++--
 src/controls/ListSectionHeader.qml | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/controls/Action.qml b/src/controls/Action.qml
index e4b850e..203b7e7 100644
--- a/src/controls/Action.qml
+++ b/src/controls/Action.qml
@@ -18,7 +18,7 @@
  */
 
 import QtQuick 2.7
-import QtQuick.Controls 2.5 as Controls
+import QtQuick.Controls 2.4 as Controls
 import "private"
 
 /**
diff --git a/src/controls/ActionToolBar.qml b/src/controls/ActionToolBar.qml
index 81324a4..d9299d5 100644
--- a/src/controls/ActionToolBar.qml
+++ b/src/controls/ActionToolBar.qml
@@ -19,7 +19,7 @@
 
 import QtQuick 2.7
 import QtQuick.Layouts 1.2
-import QtQuick.Controls 2.5 as Controls
+import QtQuick.Controls 2.4 as Controls
 import org.kde.kirigami 2.5 as Kirigami
 import "private"
 
@@ -172,7 +172,7 @@ Item {
 
             kirigamiAction: Kirigami.Action {
                 icon.name: "overflow-menu"
-                children: Array.prototype.map.call(root.actions, i => i).concat(Array.prototype.map.call(hiddenActions, i => i))
+                children: Array.prototype.map.call(root.actions, function (i) { return i }).concat(Array.prototype.map.call(hiddenActions, function (i) { return i }))
             }
 
             menu.submenuComponent: ActionsMenu {
diff --git a/src/controls/ListSectionHeader.qml b/src/controls/ListSectionHeader.qml
index 1309fc3..c0a726d 100644
--- a/src/controls/ListSectionHeader.qml
+++ b/src/controls/ListSectionHeader.qml
@@ -19,7 +19,7 @@
 
 import QtQuick 2.5
 import QtQuick.Layouts 1.2
-import QtQuick.Controls 2.5 as QQC2
+import QtQuick.Controls 2.4 as QQC2
 import org.kde.kirigami 2.10
 
 /**
